pkg/auth/dependency/newinteraction/nodes/select_identity_oauth.go

```diff
@@ -2,46 +2,122 @@ package nodes
 
 import (
 	"errors"
+	"net/http"
 
 	"github.com/authgear/authgear-server/pkg/auth/config"
 	"github.com/authgear/authgear-server/pkg/auth/dependency/identity"
 	"github.com/authgear/authgear-server/pkg/auth/dependency/newinteraction"
 	"github.com/authgear/authgear-server/pkg/auth/dependency/sso"
 	"github.com/authgear/authgear-server/pkg/core/authn"
+	"github.com/authgear/authgear-server/pkg/core/crypto"
+	"github.com/authgear/authgear-server/pkg/core/skyerr"
 )
 
 func init() {
-	newinteraction.RegisterNode(&NodeSelectIdentityOAuth{})
+	newinteraction.RegisterNode(&NodeSelectIdentityOAuthProvider{})
+	newinteraction.RegisterNode(&NodeSelectIdentityOAuthUserInfo{})
 }
 
-type InputSelectIdentityOAuth interface {
-	GetUserInfo() sso.AuthInfo
+var ErrOAuthProviderNotFound = skyerr.NotFound.WithReason("OAuthProviderNotFound").New("oauth provider not found")
+
+type InputSelectIdentityOAuthProvider interface {
+	GetProviderAlias() string
+	GetState() string
+	GetNonceSource() *http.Cookie
+	GetErrorRedirectURI() string
 }
 
-type EdgeSelectIdentityOAuth struct {
+type EdgeSelectIdentityOAuthProvider struct {
 	Config config.OAuthSSOProviderConfig
 }
 
-func (e *EdgeSelectIdentityOAuth) Instantiate(ctx *newinteraction.Context, graph *newinteraction.Graph, rawInput interface{}) (newinteraction.Node, error) {
-	input, ok := rawInput.(InputSelectIdentityOAuth)
+type NodeSelectIdentityOAuthProvider struct {
```

Comment:
I think it's better not to interleave the type definitions here?

Reply:
Do you mean separating `SelectOAuthProvider` and `SelectOAuthUserInfo` into 2 files? Or something else?

Reply:
I mean the layout of file, now it is:
1. define `EdgeSelectIdentityOAuth`
2. define `NodeSelectIdentityOAuthProvider`
3. define methods of `EdgeSelectIdentityOAuth`
4. define methods of `NodeSelectIdentityOAuthProvider`

3 should be moved before 2.

Separating the `SelectOAuthProvider` and `SelectOAuthUserInfo` into 2 files also seems better, since both contains large amount of non-trivial and de-coupled logic.